io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus3em/IoShellyPlus3em.java

```diff
@@ -0,0 +1,146 @@
+package io.openems.edge.io.shelly.shellyplus3em;
+
+import org.osgi.service.event.EventHandler;
+
+import io.openems.common.channel.AccessMode;
+import io.openems.common.channel.Level;
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.types.OpenemsType;
+import io.openems.edge.common.channel.BooleanDoc;
+import io.openems.edge.common.channel.BooleanWriteChannel;
+import io.openems.edge.common.channel.Doc;
+import io.openems.edge.common.channel.StateChannel;
+import io.openems.edge.common.channel.value.Value;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.io.api.DigitalOutput;
+import io.openems.edge.meter.api.ElectricityMeter;
+import io.openems.edge.meter.api.SinglePhaseMeter;
+
+public interface IoShellyPlus3em
+		extends DigitalOutput, SinglePhaseMeter, ElectricityMeter, OpenemsComponent, EventHandler {
+
+	public static enum ChannelId implements io.openems.edge.common.channel.ChannelId {
+		/**
+		 * Holds writes to Relay Output for debugging.
+		 *
+		 * <ul>
+		 * <li>Interface: ShellyPlug
+		 * <li>Type: Boolean
+		 * <li>Range: On/Off
+		 * </ul>
+		 */
+		DEBUG_RELAY(Doc.of(OpenemsType.BOOLEAN)), //
+		/**
+		 * Relay Output.
+		 *
+		 * <ul>
+		 * <li>Interface: ShellyPlug
+		 * <li>Type: Boolean
+		 * <li>Range: On/Off
+		 * </ul>
+		 */
+		RELAY(new BooleanDoc() //
+				.accessMode(AccessMode.READ_WRITE) //
+				.onChannelSetNextWriteMirrorToDebugChannel(ChannelId.DEBUG_RELAY)),
+
+		/**
+		 * Indicates if an update is available.
+		 *
+		 * <ul>
+		 * <li>Interface: ShellyPlug
+		 * <li>Type: Boolean
+		 * <li>Level: INFO
+		 * </ul>
+		 */
+		HAS_UPDATE(Doc.of(Level.INFO) //
```

Comment:
This is an interesting feature. Should we standardize this Channel for alle Shellys? - i.e. create a `Shelly` nature that provides this information and a generic handler, that parses the `has_update` value?

Reply:
Yes, can be done, but not all shellys provide this info as a boolean. some of them just show the new versions in the responses for example ShellyPlus1PM:

`{"ble":{},"cloud":{"connected":true},"input:0":{"id":0,"state":false},"mqtt":{"connected":false},"switch:0":{"id":0, "source":"WS_in", "output":false, "apower":0.0, "voltage":230.2, "current":0.000, "aenergy":{"total":8629.000,"by_minute":[0.000,0.000,0.000],"minute_ts":1708707240},"temperature":{"tC":43.5, "tF":110.3}},"sys":{"mac":"80646FE34998","restart_required":false,"time":"17:54","unixtime":1708707267,"uptime":491734,"ram_size":260248,"ram_free":108996,"fs_size":458752,"fs_free":143360,"cfg_rev":21,"kvs_rev":2,"schedule_rev":0,"webhook_rev":0,"**available_updates":{"stable":{"version":"1.2.1"}}**,"reset_reason":3},"wifi":{"sta_ip":"192.168.178.169","status":"got ip","ssid":"heizung","rssi":-50,"ap_client_count":0},"ws":{"connected":false}}`